src/lightgbm/src/main/scala/TrainParams.scala

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+/** Defines the common Booster parameters passed to the LightGBM learners.
+  *
+  * @param numIterations Number of iterations, LightGBM constructs num_class * num_iterations trees.
+  * @param learningRate Learning rate or shrinkage rate.
+  * @param numLeaves Number of leaves.
+  */
+abstract class TrainParams(val parallelism: String, val numIterations: Int, val learningRate: Double,
+                           val numLeaves: Int) extends Serializable {
+  override def toString(): String = {
+    s"is_pre_partition=True boosting_type=gbdt tree_learner=$parallelism num_iterations=$numIterations " +
+      s"learning_rate=$learningRate num_leaves=$numLeaves"
+  }
+}
+
+/** Defines the Booster parameters passed to the LightGBM classifier.
+  */
+class ClassifierTrainParams(parallelism: String, numIterations: Int, learningRate: Double, numLeaves: Int)
+  extends TrainParams(parallelism, numIterations, learningRate, numLeaves) {
+  override def toString(): String = {
+    s"objective=binary metric=binary_logloss,auc ${super.toString()}"
+  }
+}
+
+object ClassifierTrainParams {
```

Comment:
Unless I'm mistaken, you could get the same effect by making `ClassifierTrainParams` into a case-class.

Reply:
done